nwbwidgets/controllers/basecontroller.py

```diff
@@ -0,0 +1,64 @@
+from typing import Dict
+
+import ipywidgets as widgets
+
+
+class BaseController(widgets.Box):
```

Comment:
I tried several ways of doing a mixin with `ABC` to use `abstractmethod`, but ran into metaclass conflicts with the `widgets.Box` so simply opted to raise `NotImplementedErrors` instead